CRM/Contribute/Import/Parser/Contribution.php

```diff
@@ -684,12 +684,14 @@ private function deprecatedFormatParams($params, &$values, $create = FALSE) {
               $params['contribution_contact_id'] = $matchingContactIds[0];
             }
           }
-          elseif (!empty($params['contribution_id']) || !empty($params['trxn_id']) || !empty($params['invoice_id'])) {
+          elseif (!empty($params['id']) || !empty($params['trxn_id']) || !empty($params['invoice_id'])) {
             // when update mode check contribution id or trxn id or
             // invoice id
+            // @todo - this check is obsolete. It survives for now
+            // in order to keep the rc patch small & non-conflicty.
             $contactId = new CRM_Contribute_DAO_Contribution();
-            if (!empty($params['contribution_id'])) {
-              $contactId->id = $params['contribution_id'];
+            if (!empty($params['id'])) {
+              $contactId->id = $params['id'];
```

Comment:
@eileenmcnaughton ok this seems to make sense but the $variable name is very confusing just FYI

Reply:
Yep - once merged to master most of that can be removed (which was the original point of the test)